reservations/allocation_data_builder.py

```diff
@@ -0,0 +1,134 @@
+import datetime
+from typing import Dict, List
+
+from django.utils import timezone
+
+from applications.models import ApplicationRound
+from reservation_units.models import ReservationUnit
+from reservations.allocation_models import (
+    AllocationBasket,
+    AllocationData,
+    AllocationEvent,
+    AllocationSpace,
+)
+
+
+class AllocationDataBuilder(object):
+    def __init__(
+        self, application_round: ApplicationRound, included_baskets: [int] = []
+    ):
+        self.period_start: datetime.date = application_round.reservation_period_begin
+        self.period_end: datetime.date = application_round.reservation_period_end
+        self.application_round = application_round
+        self.included_baskets = included_baskets
+        self.baskets = {}
+
+    def get_allocation_data(self):
+        spaces: dict[int, AllocationSpace] = {}
+        for unit in self.application_round.reservation_units.all():
+            space = self.get_space(unit=unit)
+            spaces[space.id] = space
+        allocation_events = []
+        event_baskets = self.get_event_baskets()
+
+        for application in self.application_round.applications.all():
+            for application_event in application.application_events.all():
+                space_ids = list(
+                    map(
+                        lambda x: x.reservation_unit.id,
+                        application_event.event_reservation_units.all(),
+                    )
+                )
+                allocation_events.append(
+                    AllocationEvent(
+                        id=application_event.id,
+                        occurrences=application_event.get_all_occurrences(),
+                        period_start=self.period_start,
+                        period_end=self.period_end,
+                        baskets=event_baskets.get(application_event.id),
+                        space_ids=space_ids,
+                        begin=application_event.begin,
+                        end=application_event.end,
+                        min_duration=application_event.min_duration,
+                        max_duration=application_event.max_duration
+                        if application_event.max_duration is not None
```

Comment:
do we need explicit `None` check here? 
For example using  `if application_event.max_duration` to also check against zero (0) value => so what if the max_duration is zero, should it be count or not as max_duration?

Reply:
There is validation that checks that max duration is none or larger than min duration, so there should be no need to check for zeros.